cmd/kubectl-gadget/utils/trace.go

```diff
@@ -287,44 +430,231 @@ RetryLoop:
+	if !config.CommonFlags.AllNamespaces {
+		namespace = GetNamespace()
+	}
+
+	filter := map[string]string{
+		"gadgetName":    config.GadgetName,
+		"nodeName":      config.CommonFlags.Node,
+		"namespace":     namespace,
+		"podName":       config.CommonFlags.Podname,
+		"containerName": config.CommonFlags.Containername,
+		"outputMode":    config.TraceOutputMode,
+	}
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: labelsFromFilter(filter),
+	}
+
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return []gadgetv1alpha1.Trace{}, err
+	}
+
+	return traces.Items, nil
+}
+
+// PrintAllTraces prints all traces corresponding to the given config.CommonFlags.
+func PrintAllTraces(config *TraceConfig) error {
+	traces, err := getTraceListFromParameters(config)
+	if err != nil {
+		return err
+	}
+
+	type printingInformation struct {
+		namespace     string
+		nodeName      string
+		podname       string
+		containerName string
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+
+	fmt.Fprintln(w, "NAMESPACE\tNODE(S)\tPOD\tCONTAINER\tTRACEID")
+
+	printingMap := map[string]*printingInformation{}
+
+	for _, trace := range traces {
+		id, present := trace.ObjectMeta.Labels[GLOBAL_TRACE_ID]
+		if !present {
+			continue
+		}
+
+		node := trace.Spec.Node
+
+		_, present = printingMap[id]
+		if present {
+			if node == "" {
+				continue
+			}
+
+			// If an entry with this traceID already exists, we just update the node
+			// name by concatenating it to the string.
+			printingMap[id].nodeName = printingMap[id].nodeName + "," + node
+		} else {
+			// Otherwise, we simply create a new entry.
+			filter := trace.Spec.Filter
+
+			printingMap[id] = &printingInformation{
+				namespace:     filter.Namespace,
+				nodeName:      node,
+				podname:       filter.Podname,
+				containerName: filter.ContainerName,
+			}
 		}
-		genericStreamsDisplay(contextLogger, client, params, &results, transformLine)
-	} else {
-		customResultsDisplay(contextLogger, nodes, &results)
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	for id, info := range printingMap {
+		fmt.Fprintf(w, "%v\t%v\t%v\t%v\t%v\n", info.namespace, info.nodeName, info.podname, info.containerName, id)
+	}
+
+	w.Flush()
+
+	return nil
+}
+
+// RunTraceAndPrintStream creates a trace, prints its output and deletes
+// it.
+// It equals calling separately CreateTrace(), then PrintTraceOutputFromStream()
+// and DeleteTrace().
+// This function is thought to be used with "one-run" gadget, i.e. gadget
+// which runs a trace when it is created.
+func RunTraceAndPrintStream(config *TraceConfig, transformLine func(string) string) error {
+	traceID, err := CreateTrace(config)
+	if err != nil {
+		return fmt.Errorf("error creating trace: %w", err)
+	}
+
+	defer DeleteTrace(traceID)
+
+	if config.TraceOutputMode != "Stream" {
+		return errors.New("TraceOutputMode must be Stream. Otherwise, call RunTraceAndPrintStatusOutput!")
+	}
```

Comment:
What about doing this before creating the Trace? I think it is not necessary to create it if we are going to delete it because the `TraceOutputMode` is not compatible with this function. WDYT?